azure/README.md

```diff
@@ -0,0 +1,197 @@
+# Setup your secrets in Azure
+
+In this setup we integrate the secrets exercise with Azure AKS and let pods consume secrets from an Azure Key Vault. If you want to know more about integrating secrets with AKS, check [this link](https://github.com/Azure/secrets-store-csi-driver-provider-azure).
+Please make sure that the account in which you run this exercise has either Log Analytics enabled, or is not linked to your current subscriptions and/or DTAP environment.
+
+## Pre-requisites
+
+Have the following tools installed:
+
+-   az CLI - [Installation](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)
+-   Tfenv (Optional) - [Installation](https://github.com/tfutils/tfenv)
+-   Terraform CLI - [Installation](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+-   Wget - [Installation](https://www.jcchouinard.com/wget/)
+-   Helm [Installation](https://helm.sh/docs/intro/install/)
+-   Kubectl [Installation](https://kubernetes.io/docs/tasks/tools/)
+-   jq [Installation](https://stedolan.github.io/jq/download/)
+
+Make sure you have an active subscription at Azure for which you have configured the credentials on the system where you will execute the steps below.
+
+Please note that this setup relies on bash scripts that have been tested in MacOS and Linux. We have no intention of supporting vanilla Windows at the moment.
+
+## Installation
+
+**Note-I**: We create resources in `east us` by default. You can set the region by editing `terraform.tfvars`.
+
+**Note-II**: The cluster you create has its access bound to the public IP of the creator. In other words: the cluster you create with this code has its access bound to your public IP-address if you apply it locally. If you switched to a different network, you'll need to run `terraform apply` again to update the firewall rules.
+
+### (Optional) Multi-user setup: shared state
+
+If you want to host a multi-user setup, you will probably want to share the state file so that everyone can try related challenges. We have provided a starter to easily do so using an Azure storage container.
+
+First, enable the `Microsoft.Storage` API (if it isn't already) using:
+
+```bash
+az provider register --namespace Microsoft.Storage
+```
+
+Then, apply the Terraform (optionally add `-var="region=YOUR_DESIRED_REGION"` to the apply to use a region other than the default `East US`):
+
+```bash
+cd shared-state
+terraform init
+terraform apply
+```
+
+The storage account name should be in the output. Please use that to configure the Terraform backend in `main.tf` by uncommenting the part on the `backend "azurerm"`.
+
+**Note**: You'll need to follow the description below for the "existing resource group" i.e., use the `data.azurerm_resource_group.default` resource.
```

Comment:
This is rather unclear to me: what do i have to do? Maybe make it clear that this is noted in step1 below?

Reply:
Maybe better now?